GitUI/UserControls/CaseSensitiveComboBox.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.ComponentModel;
+using System.Windows.Forms;
+using GitCommands.UserRepositoryHistory;
+
+namespace GitUI.UserControls
+{
+    public class CaseSensitiveComboBox : ComboBox
+    {
+        private const AutoCompleteMode Mode = AutoCompleteMode.SuggestAppend;
+        private string _lastTextChangedValue;
+
+        private bool SystemAutoCompleteEnabled
+        {
+            get => Mode != AutoCompleteMode.None && DropDownStyle != ComboBoxStyle.DropDownList;
+        }
+
+        protected override void OnValidating(CancelEventArgs e)
+        {
+            if (SystemAutoCompleteEnabled)
+            {
+                NotifyAutoComplete();
+            }
+        }
+
+        protected override void OnTextChanged(EventArgs e)
+        {
+            if (SystemAutoCompleteEnabled)
+            {
+                string text = Text;
+                if (text == _lastTextChangedValue)
+                {
+                    return;
+                }
+
+                base.OnTextChanged(e);
+                _lastTextChangedValue = text;
+            }
+            else
+            {
+                base.OnTextChanged(e);
+            }
```

Comment:
If the sequence does not matter, this can be simplified:
```suggestion
                _lastTextChangedValue = text;
            }

            base.OnTextChanged(e);
```